lib/voyage/templates/concerns_analytics_track.rb

```diff
@@ -0,0 +1,51 @@
+# :nocov:
+module AnalyticsTrack
+  extend ActiveSupport::Concern
+
+  # rubocop:disable Metrics/LineLength
+  # Example Traditional Event: analytics_track(user, 'Created Widget', { widget_name: 'foo' })
+  # Example Page View:         analytics_track(user, 'Page Viewed', { page_name: 'Terms and Conditions', url: '/terms' })
+  #
+  # NOTE: setup some defaults that we want to track on every event mixpanel_track
+  # NOTE: the identify step happens on every page load to keep intercom.io and mixpanel people up to date
+  # rubocop:enable Metrics/LineLength
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def analytics_track(user, event_name, options = {})
+    return if user.tester?
+
+    sanitized_options = sanitize_hash_javascript(options)
+
+    # rubocop:disable Style/RescueModifier, Lint/RescueWithoutErrorClass
+    segment_attributes = {
+      user_id: user.uuid,
+      event: event_name,
+      properties: {
+        browser: (browser.name rescue 'unknown'),
+        browser_id: (browser.id rescue 'unknown'),
+        browser_version: (browser.version rescue 'unknown'),
+        platform: (browser.platform rescue 'unknown'),
+        roles: (user.roles.map(&:to_s).join(',') rescue ''),
+        rails_env: Rails.env.to_s,
+      }.merge(sanitized_options),
```

Comment:
@noahsettersten this was on my list to clean up in another project, the rescues just seem so ugly; probably not in scope here though.

Reply:
I agree, these rescues are holdovers from early on in Voyage but they were moved into a separate concern in this PR. Definitely something to address, I'll create a separate issue to keep track of it: #35.